File: 383-ransom-note/383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        # for each char in ransomNote
        # If it is in magazine:
            # Remove it from magazine.
        # Else:
            #Return False
        # Return True
        for i in ransomNote:
            if i in magazine:
                magazine = magazine.replace(i, '', 1)
                ransomNote = ransomNote.replace(i, '', 1)
            else:
                return False
        return True
        

File: 383-ransom-note/383-ransom-note.js
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function(ransomNote, magazine) {
    while (ransomNote.length > 0){
        const nextLetter = ransomNote.substring(0, 1);
        const indexInMagazine = magazine.indexOf(nextLetter);
        
        if(indexInMagazine > -1) {
            magazine = magazine.substring(0, indexInMagazine) + magazine.substring(indexInMagazine + 1);
        } else {
            return false;
        }
        ransomNote = ransomNote.substring(1);
    }
    return true;
};